### PR TITLE
Added Date Sample required (Y/N) setting in Setup

### PR DESCRIPTION
**Pull request by xispa**
## Description of the issue/feature this PR addresses

This Pull Request adds a setting in setup that allows the user to make the "DateSampled" field required or not on sample registration. This differs from when the sampling workflow is enabled in the sense that no information about the sampling (e.g. like sampler) is required. Quite often clinic/hospital labs might need to register a sample for a given patient beforehand. This is, when sample has not yet been collected or is about to be collected. These labs are not interested on capturing the sampler and/or sampling process information.

Captura de 2024-08-21 18-57-55

If non-required and date sampled is empty, the sample won't transition automatically, but will remain in "sample due" status and a viewlet will be displayed at the top of the sample view stating the sample cannot be received at the lab because collection date is not set.

Captura de 2024-08-21 18-53-27

## Current behavior before PR

Is not possible to create samples without date sampled unless Sampling workflow enabled

## Desired behavior after PR is merged

Is possible to create samples without date sampled if setup setting is not checked

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

File: src/bika/lims/content/bikasetup.py [hunk 1]
```diff
@@ -575,6 +575,23 @@ def getCounterTypes(self, instance=None):
...
+                default="Date sampled required"
+            ),
+            description=_(
+                "description_bikasetup_date_sampled_required",
+                default="Select this to make DateSampled field required on "
+                        "sample creation. This functionality only takes "
+                        "effect when 'Sampling workflow' is not active"
+            ),
+        ),
+    ),
```
Comment (ramonski) [inline src/bika/lims/content/bikasetup.py:595]: Please add the counterpart for this setting in the new SENAITE setup

**Review by ramonski — CHANGES_REQUESTED** [review]
When the setting "Date sampled required" is disabled, the field is hidden from the sample registration form.

However, if an existing sample is copied, the "Date Sampled" is kept (but not displayed) and get set for the new sample on creation.

Therefore, if the "Auto-receive samples" is enabled, the sample is automatically received with that date and there is no way to get it back to "Sample registered" state.

Suggestion: Leave the field visible in the add form but flush the value.

**Comment by xispa** [comment]
> When the setting "Date sampled required" is disabled, the field is hidden from the sample registration form

I cannot reproduce this. The system only hides "DateSampled" when the sampling workflow is enabled, regardless of the value for the setting "Date sampled required". This is done here: https://github.com/senaite/senaite.core/pull/2609/files#diff-4211a506277eb5b6580751781aa5d57867f7e1ef6a17c09f6e40da47e6776df5R175

Am I missing something?

> However, if an existing sample is copied, the "Date Sampled" is kept (but not displayed) and get set for the new sample on creation.

If the sampling workflow is enabled, then the system should not copy the value of "DateSampled" to the new object. Well noted, will open another PR for this.

File: src/bika/lims/content/bikasetup.py (hunk 1 above)
Comment (xispa) [reply src/bika/lims/content/bikasetup.py:595]: Addressed with https://github.com/senaite/senaite.core/pull/2609/commits/556e8bb5e4b2c2d51720b2cb68adf03dbd3739e8

**Review by ramonski — APPROVED** [review]
Great, thanks!

**Pull request merged**